packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,234 @@
+			forest.tryMoveCursorTo(forest.root(forest.rootField), this.cursor);
+		} else {
+			this.cursor = _cursor.fork();
+		}
+	}
+
+	public tryMoveDown = (key: FieldKey):
+		{ result: TreeNavigationResult; isArray: boolean; } => {
+		if (key !== EmptyKey && this.cursor.length(EmptyKey) && isNaN(Number(key))) {
+			return { result: TreeNavigationResult.NotFound, isArray: true };
+		}
+		let result = this.cursor.down(key, 0);
+		let isArray = false;
+		if (result === TreeNavigationResult.NotFound) {
+			// reading an array
+			result = this.cursor.down(EmptyKey, Number(key));
+		} else {
+			isArray = !!this.cursor.length(EmptyKey);
+		}
+		return { result, isArray };
+	};
+
+	public getDummyArray = (length: number): undefined[] => {
+		const dummy: undefined[] = [];
+		for (let i = 0; i < length; i++) {
+			dummy.push(undefined);
+		}
+		return dummy;
+	};
+
+	public getNodeData = <T>(cursor?: ITreeSubscriptionCursor): TreeValue | EditableTreeNode<T> | undefined => {
+		const _cursor = cursor ?? this.cursor;
+		if (_cursor.value !== undefined) {
+			const result: TreeValue = _cursor.value;
+			return result;
+		}
+		return proxify<T>(this.forest, _cursor);
+	};
+
+	public getArrayGreedy = <T>() => {
+		const getNodeData = this.getNodeData;
+		return mapCursorField(this.cursor, EmptyKey,
+			(cursor: ITreeCursor): TreeValue | EditableTreeNode<T> | undefined => {
+			return getNodeData(cursor as ITreeSubscriptionCursor);
+		});
+	};
+
+	[getTypeSymbol](key?: FieldKey): TreeSchemaIdentifier {
+		if (key) {
+			const { result } = this.tryMoveDown(key);
+			if (result === TreeNavigationResult.Ok) {
+				const type = this.cursor.type;
+				this.cursor.up();
+				return type;
+			}
+			return brand("");
+		}
+		return this.cursor.type;
+	}
+}
+
+/**
+ * A Proxy handler together with a {@link ProxyTarget} implements a basic read/write access to the Forest
+ * by means of the cursors.
+ */
+const handler: ProxyHandler<ProxyTarget> = {
+	get: (target: ProxyTarget, key: string | symbol, receiver: ProxyTarget): unknown => {
+		if (typeof key === "string") {
+			const { result, isArray } = target.tryMoveDown(brand(key));
+			if (result === TreeNavigationResult.Ok) {
+				const node = target.getNodeData();
+				target.cursor.up();
+				return node;
+			} else if (isArray) {
+				switch (key) {
+					case "length":
+						return target.cursor.length(EmptyKey);
+					default:
+						return [][key as keyof []];
+				}
+			}
+		}
+		if (key === Symbol.iterator) {
+			const data = target.getArrayGreedy();
+			return data[Symbol.iterator];
+		} else if (key === getTypeSymbol) {
+			return target[getTypeSymbol].bind(target);
+		}
+		return undefined;
+	},
+	set: (target: ProxyTarget, key: string, value: unknown, receiver: ProxyTarget): boolean => {
+		throw new Error("Not implemented.");
+	},
+	deleteProperty: (target: ProxyTarget, key: string): boolean => {
+		throw new Error("Not implemented.");
+	},
+	has: (target: ProxyTarget, key: string | symbol): boolean => {
```

Comment:
I just learned that apparently you can't just user proxy to implement any API you want regardless of the target type because of these invariants for "has": https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/has#invariants

I guess we have to follow those which makes what properties exist on the target matter.
Maybe we should just return true for all strings to avoid violating this. Either that or we have to make the target only have symbols (Which I think you did before, but I wrongly though was pointless)

Reply:
I am not sure that it matters since
a) we don't define non-configurable properties neither define a target to be not extensible
b) as we expose the proxy via a public API type, such exception should never occur as it is hidden behind the API

The weak point I see currently is that we have to always keep the target's type and the API in sync semantically, what's always not as good as a physically implemented contract, imho, but I think that might be fine for a while.